packages/suite/src/reducers/suite/suiteReducer.ts

```diff
@@ -432,30 +435,64 @@ export const selectHasExperimentalFeature =
     (feature: ExperimentalFeature) => (state: SuiteRootState) =>
         state.suite.settings.experimental?.includes(feature) ?? false;
 
-export const selectIsFirmwareRevisionCheckEnabledAndFailed = (
-    state: SuiteRootState & DeviceRootState & MessageSystemRootState,
-) => {
+type StateForFirmwareChecks = SuiteRootState & DeviceRootState & MessageSystemRootState;
+
+/**
+ * Get firmware revision check error, or null if check was successful / skipped.
+ */
+export const selectFirmwareRevisionCheckError = (state: StateForFirmwareChecks) => {
+    const device = selectDevice(state);
+    if (!isDeviceAcquired(device) || !device.authenticityChecks) return null;
+
     const { isFirmwareRevisionCheckDisabled } = state.suite.settings;
-    const isFirmwareRevisionCheckDisabledByMessageSystem = selectIsFeatureDisabled(
-        state,
-        Feature.firmwareRevisionCheck,
-    );
+    const isDisabledByMessage = selectIsFeatureDisabled(state, Feature.firmwareRevisionCheck);
+    const isCheckEnabled = !isFirmwareRevisionCheckDisabled && !isDisabledByMessage;
+    const checkResult = device.authenticityChecks.firmwareRevision; // null means not performed, then don't consider it failed
 
-    if (isFirmwareRevisionCheckDisabled || isFirmwareRevisionCheckDisabledByMessageSystem) {
-        return false;
-    }
+    return isCheckEnabled && checkResult?.success === false ? checkResult.error : null;
+};
+
+/**
+ * Determine hard failure of firmware revision check - specific error types which are severe.
+ * If Suite is offline and cannot perform check or there is some unexpected error, a banner is shown but device is accessible.
+ */
+const selectIsFirmwareRevisionCheckEnabledAndFailed = (state: StateForFirmwareChecks): boolean => {
+    const error = selectFirmwareRevisionCheckError(state);
 
+    return error ? !['cannot-perform-check-offline', 'other-error'].includes(error) : false;
+};
```

Comment:
Took me a long while of iterating and making a mess until I came up with the best way:
split `selectFirmwareRevisionCheckError` from `selectIsFirmwareRevisionCheckEnabledAndFailed` instead of having one.
See usages of `selectFirmwareRevisionCheckError`, and hash respectively. In all those cases I'd have a lot of boilerplate, potentially making a mistake... and such mistake is currently in develop, see comment in TODO